db/migrate/20160527140046_foreign_keys.rb

```diff
@@ -0,0 +1,116 @@
+
+    add_foreign_key :field_guides, :projects, on_update: :cascade, on_delete: :cascade
+
+    add_foreign_key :memberships, :user_groups, on_update: :cascade, on_delete: :cascade
+    add_foreign_key :memberships, :users, on_update: :cascade, on_delete: :cascade
+
+    add_foreign_key :oauth_access_grants, :users, column: :resource_owner_id, on_update: :cascade, on_delete: :cascade
+    add_foreign_key :oauth_access_grants, :oauth_applications, column: :application_id, on_update: :cascade, on_delete: :cascade
+
+    add_foreign_key :oauth_access_tokens, :users, column: :resource_owner_id, on_update: :cascade, on_delete: :cascade
+    add_foreign_key :oauth_access_tokens, :oauth_applications, column: :application_id, on_update: :cascade, on_delete: :cascade
+
+    ProjectContent.joins("LEFT OUTER JOIN projects ON projects.id = project_contents.project_id").where("projects.id IS NULL").delete_all
+    add_foreign_key :project_contents, :projects, on_update: :cascade, on_delete: :cascade
+
+    ProjectPage.joins("LEFT OUTER JOIN projects ON projects.id = project_pages.project_id").where("projects.id IS NULL").delete_all
+    add_foreign_key :project_pages, :projects, on_update: :cascade, on_delete: :cascade
+
+    # add_foreign_key :recents, :classifications
+    # add_foreign_key :recents, :subjects
+
+    SetMemberSubject.joins("LEFT OUTER JOIN subject_sets ON subject_sets.id = set_member_subjects.subject_set_id").where("subject_sets.id IS NULL").delete_all
+    SetMemberSubject.joins("LEFT OUTER JOIN subjects ON subjects.id = set_member_subjects.subject_id").where("subjects.id IS NULL").delete_all
+    add_foreign_key :set_member_subjects, :subject_sets, on_update: :cascade, on_delete: :cascade
+    add_foreign_key :set_member_subjects, :subjects, on_update: :cascade, on_delete: :cascade
+
+    SubjectQueue.joins("LEFT OUTER JOIN subject_sets ON subject_sets.id = subject_queues.subject_set_id").where("subject_sets.id IS NULL").delete_all
+    add_foreign_key :subject_queues, :users, on_update: :cascade, on_delete: :cascade
+    add_foreign_key :subject_queues, :workflows, on_update: :cascade, on_delete: :cascade
+    add_foreign_key :subject_queues, :subject_sets, on_update: :cascade, on_delete: :restrict
+
+    add_foreign_key :subject_sets, :projects, on_update: :cascade, on_delete: :cascade
+
+    # add_foreign_key :subject_sets_workflows, :subject_sets
+    # add_foreign_key :subject_sets_workflows, :workflows
+
+    # add_foreign_key :subject_workflow_counts, :workflows
+    # add_foreign_key :subject_workflow_counts, :subjects
+
+    Subject.joins("LEFT OUTER JOIN projects ON projects.id = subjects.project_id").where("projects.id IS NULL").update_all(project_id: nil)
+    add_foreign_key :subjects, :projects, on_update: :cascade, on_delete: :restrict
+    add_foreign_key :subjects, :users, column: :upload_user_id, on_update: :cascade, on_delete: :nullify
```

Comment:
User is `activatable` anyways, so we shouldn't ever delete them. I'll change this to a `restrict`.